pkg/ds/daemon_set.go

```diff
@@ -315,12 +329,21 @@ func (ds *daemonSet) removePods() error {
 	toUnscheduleSorted := types.NewNodeSet(currentNodes...).Difference(types.NewNodeSet(eligible...)).ListNodes()
 	ds.logger.NoFields().Infof("Need to unschedule %d nodes", len(toUnscheduleSorted))
 
+	if len(toUnscheduleSorted) > 0 {
+		ds.CancelReplication()
+	}
+
 	for _, node := range toUnscheduleSorted {
 		err := ds.unschedule(node)
 		if err != nil {
 			return util.Errorf("Error unscheduling node: %v", err)
 		}
 	}
+
+	if len(podLocations)-len(toUnscheduleSorted) > 0 {
+		return ds.PublishToReplication()
```

Comment:
right, but `addPods()` gets called right after `removePods()` in `WatchDesires()` so you shouldn't need it here right?